File: LeetCode/Strings/CountAndSay.java
package LeetCode.Strings;

public class CountAndSay {
    public static String countAndSay(int n) {
        if (n == 1) {
            return "1";
        }
        String say = countAndSay(n - 1);
        String res = "";
        for (int i = 0; i < say.length(); i++) {
            char c = say.charAt(i);
            int count = 1;
            while (i < say.length() - 1 && say.charAt(i) == say.charAt(i + 1)) {
                count++;
                i++;
            }
            res += String.valueOf(count) + String.valueOf(c);
        }
        return res;
    }

    public static void main(String[] args) {
        int n = 4;
        System.out.println(countAndSay(n));
    }
}

File: LeetCode/Strings/CountAndSay.js
function countAndSay(n) {
  if (n === 1) {
    return "1"; // Base case
  }

  const say = countAndSay(n - 1);
  let result = "";
  let count = 1;

  for (let i = 0; i < say.length; i++) {
    if (say[i] === say[i + 1]) {
      count++;
    } else {
      result += count + say[i];
      count = 1;
    }
  }

  return result;
}

// Example usage
const n = 5;
const result = countAndSay(n);
console.log(result); // Output: "111221"
